workflows/plugins/include/utils/alerts.py

```diff
@@ -35,12 +37,17 @@ def task_failure_alert(context):
     )
 
 
-def dag_failure_callback(context):
-    """Callback function for DAG failure."""
-    try:
-        task_failure_alert(context)
-    except Exception as e:
-        logger.error(f"Error in sending alert: {e}")
+class DagErrorNotifier(BaseNotifier):
```

Comment:
I would create 2 of them, one  that has the collection and also sets the status, like you do here,  and one simple which is just sending alerts without any collection. It would be easier in case we want to extend in the future